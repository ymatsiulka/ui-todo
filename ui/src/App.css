.App {
  text-align: center;
}

.App-logo {
  height: 40vmin;
  pointer-events: none;
}

@media (prefers-reduced-motion: no-preference) {
  .App-logo {
    animation: App-logo-spin infinite 20s linear;
  }
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }

  to {
    transform: rotate(360deg);
  }
}

.todo-content {
  height: 100vh;
  display: flex;
  flex-direction: column;
}

.top-layout {
  background-image: linear-gradient(to right, rgba(9, 5, 47, .65) 0%, rgba(80, 29, 94, .85) 100%), url('./assets/mountains.jpg');
  background-position: center;
  background-size: cover;

  height: 35%;
  display: flex;
  justify-content: center;
  align-items: center;
}

.bottom-layout {
  height: 65%;
}

.todo-header-text {
  color: white;
}

.todo-container {
  width: 30%
}

.todo-selector {

  margin-top: 10px;
}

.todo-header-text {}

.bottom-layout {
  transition: all 1s linear;
}

.b-gray {
  background-color: gray;
}

.checkbox-container {
  display: flex;
}

.checkbox-label {
  margin-left: 7.5px;
}