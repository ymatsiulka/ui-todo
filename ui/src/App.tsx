import React from 'react';
import { Provider } from 'react-redux';
import store from 'store/store';
import styles from './App.m.scss';
import Todo from 'modules/pages/Todo/Todo';

const App: React.FC = () => {
  return (
    <Provider store={store}>
      <div className={styles.container}>
        <Todo />
      </div>
    </Provider>
  );
};

export default App;
