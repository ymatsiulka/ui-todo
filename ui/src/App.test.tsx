import React from 'react';
import { render } from '@testing-library/react';
import App from 'App';

test('App should render properly', async () => {
  // Arrange
  // Act
  const { asFragment } = render(<App />);

  // Assert
  expect(asFragment()).toMatchSnapshot();
});
