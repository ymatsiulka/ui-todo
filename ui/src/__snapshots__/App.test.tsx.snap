// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render properly 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div
      class="content"
    >
      <div
        class="todoContent"
      >
        <div
          bottom="10"
          style="margin-top: 10px; margin-bottom: 10px;"
          top="10"
        />
        <h1
          class="c-white"
        >
          T O D O
        </h1>
        <div
          bottom="15"
          style="margin-top: 15px; margin-bottom: 15px;"
          top="15"
        />
        <div
          class="inputWrapper"
        >
          <span
            class="check__box"
            style="margin-left: 0px; margin-right: 0px;"
          />
          <input
            class="todo-input"
            id="todo-input"
            type="text"
            value=""
          />
        </div>
        <div
          bottom="10"
          style="margin-top: 10px; margin-bottom: 10px;"
          top="10"
        />
        <ul
          class="todoList"
        />
        <div
          class="todoListActions"
        >
          <div
            class="left"
          >
            <span
              class="fs-12"
            >
              0 items left 
            </span>
          </div>
          <div
            class="middle"
          >
            <button
              class="button fs-12 c-blue"
              type="button"
            >
              <span>
                All
              </span>
            </button>
            <button
              class="button fs-12 false"
              type="button"
            >
              <span>
                Active
              </span>
            </button>
            <button
              class="button fs-12 false"
              type="button"
            >
              <span>
                Completed
              </span>
            </button>
          </div>
          <div
            class="right"
          >
            <button
              class="button fs-12"
              type="button"
            >
              <span>
                Clear Completed
              </span>
            </button>
          </div>
        </div>
        <footer />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
